python37Packages.distributed: 1.25.2 -> 1.25.3

r-ryantm opened the pull request:
Semi-automatic update generated by https://github.com/ryantm/nixpkgs-update tools. This update was made based on information from https://repology.org/metapackage/python3.7-distributed/versions.



meta.description for python37Packages.distributed is: '"Distributed computation in Python."'.


<details>
<summary>
Checks done (click to expand)
</summary>

- built on NixOS
- /nix/store/wz4rlcw5dkgzmx1xihjkw10skbf0f3lh-python3.7-distributed-1.25.3/bin/.dask-ssh-wrapped passed the binary check.
- /nix/store/wz4rlcw5dkgzmx1xihjkw10skbf0f3lh-python3.7-distributed-1.25.3/bin/dask-ssh passed the binary check.
- /nix/store/wz4rlcw5dkgzmx1xihjkw10skbf0f3lh-python3.7-distributed-1.25.3/bin/.dask-submit-wrapped passed the binary check.
- /nix/store/wz4rlcw5dkgzmx1xihjkw10skbf0f3lh-python3.7-distributed-1.25.3/bin/dask-submit passed the binary check.
- /nix/store/wz4rlcw5dkgzmx1xihjkw10skbf0f3lh-python3.7-distributed-1.25.3/bin/.dask-remote-wrapped passed the binary check.
- /nix/store/wz4rlcw5dkgzmx1xihjkw10skbf0f3lh-python3.7-distributed-1.25.3/bin/dask-remote passed the binary check.
- /nix/store/wz4rlcw5dkgzmx1xihjkw10skbf0f3lh-python3.7-distributed-1.25.3/bin/.dask-scheduler-wrapped passed the binary check.
- /nix/store/wz4rlcw5dkgzmx1xihjkw10skbf0f3lh-python3.7-distributed-1.25.3/bin/dask-scheduler passed the binary check.
- /nix/store/wz4rlcw5dkgzmx1xihjkw10skbf0f3lh-python3.7-distributed-1.25.3/bin/.dask-worker-wrapped passed the binary check.
- /nix/store/wz4rlcw5dkgzmx1xihjkw10skbf0f3lh-python3.7-distributed-1.25.3/bin/dask-worker passed the binary check.
- Warning: no invocation of /nix/store/wz4rlcw5dkgzmx1xihjkw10skbf0f3lh-python3.7-distributed-1.25.3/bin/.dask-mpi-wrapped had a zero exit code or showed the expected version
- Warning: no invocation of /nix/store/wz4rlcw5dkgzmx1xihjkw10skbf0f3lh-python3.7-distributed-1.25.3/bin/dask-mpi had a zero exit code or showed the expected version
- 10 of 12 passed binary check by having a zero exit code.
- 0 of 12 passed binary check by having the new version present in output.
- found 1.25.3 with grep in /nix/store/wz4rlcw5dkgzmx1xihjkw10skbf0f3lh-python3.7-distributed-1.25.3
- directory tree listing: https://gist.github.com/7dbafb36ef6c48a882e2e64ee55f0d17
- du listing: https://gist.github.com/0631de32e27535276996775c2faad817

</details>
<details>
<summary>
Rebuild report (if merged into master) (click to expand)
</summary>

22 total rebuild path(s)

6 package rebuild(s)

6 x86_64-linux rebuild(s)
6 i686-linux rebuild(s)
6 x86_64-darwin rebuild(s)
4 aarch64-linux rebuild(s)


First fifty rebuilds by attrpath
python27Packages.dask-jobqueue
python27Packages.dask-xgboost
python27Packages.distributed
python37Packages.dask-jobqueue
python37Packages.dask-xgboost
python37Packages.distributed

</details>

<details>
<summary>
Instructions to test this update (click to expand)
</summary>

Either download from Cachix:
```
nix-store -r /nix/store/wz4rlcw5dkgzmx1xihjkw10skbf0f3lh-python3.7-distributed-1.25.3 \
  --option binary-caches 'https://cache.nixos.org/ https://r-ryantm.cachix.org/' \
  --option trusted-public-keys '
  r-ryantm.cachix.org-1:gkUbLkouDAyvBdpBX0JOdIiD2/DP1ldF3Z3Y6Gqcc4c=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(r-ryantm's Cachix cache is only trusted for this store-path realization.)

Or, build yourself:
```
nix-build -A python37Packages.distributed https://github.com/r-ryantm/nixpkgs/archive/8107b31f86c52b483336e949a7de211a3ee4b92f.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/wz4rlcw5dkgzmx1xihjkw10skbf0f3lh-python3.7-distributed-1.25.3
ls -la /nix/store/wz4rlcw5dkgzmx1xihjkw10skbf0f3lh-python3.7-distributed-1.25.3/bin
```


</details>
<br/>


cc @teh @costrouc for testing.

r-ryantm committed:
python37Packages.distributed: 1.25.2 -> 1.25.3

Semi-automatic update generated by
https://github.com/ryantm/nixpkgs-update tools. This update was made
based on information from
https://repology.org/metapackage/python3.7-distributed/versions

Ma27 commented:
This breaks with errors like this:

```
Traceback (most recent call last):
  File "/nix/store/wz4rlcw5dkgzmx1xihjkw10skbf0f3lh-python3.7-distributed-1.25.3/bin/.dask-mpi-wrapped", line 8, in <module>
    from distributed.cli.dask_mpi import go
  File "/nix/store/wz4rlcw5dkgzmx1xihjkw10skbf0f3lh-python3.7-distributed-1.25.3/lib/python3.7/site-packages/distributed/cli/dask_mpi.py", line 4, in <module>
    from mpi4py import MPI
ModuleNotFoundError: No module named 'mpi4py'
```

I'll push a fix.

Ma27 committed:
pythonPackages.distributed: add missing dependencies

Added `mpi4py` and `bokeh` to get the executables working.

Also simplified the `lib.optional` expression as Python 3.2 and 3.4
aren't supported anymore. Rather than referencing unsupported Python 3.x
versions it's far more obvious now to only use `futures` and
`singledispatch` if Python2 is used.

costrouc commented:
@Ma27 dask distributed should not be packaged with mpi4py since it does not depend on it and is not mentioned in the requirements. If a user wants to have mpi4py with dask they need to install https://github.com/dask/dask-mpi. It looks like this package does not exist yet

costrouc commented:
I have added dask mpi in the PR reference above. Still no need for mpi4py in inputs

Ma27 commented:
As posted above, some of the binaries are failing without it. What would you suggest instead? Installing `mpi4py` globally isn't sufficient.

costrouc commented:
Yes I agree that it is right to include mpi4py since it is indeed needed and separately from dask-mpi. I opened an issue that they add it as an extra dependency https://github.com/dask/distributed/issues/2516

Ma27 commented:
Ok great! I'll run `nix-review` to ensure that this doesn't introduce more breaking packages and merge then :)
